data_encryption/models/encrypted_data.py

```diff
@@ -0,0 +1,130 @@
+        index=True,
+        help="Concerned Odoo environment (prod, preprod...)")
+    encrypted_data = fields.Binary()
+
+    _sql_constraints = [
+        ('name_environment_uniq', 'unique (name, environment)',
+         'You can not store multiple encrypted data for the same record and \
+          environment')
+    ]
+
+    @api.model
+    @ormcache('self._uid', 'name', 'env')
+    def _get(self, name, env=None):
+        if self.env.context.get('bin_size'):
+            self = self.with_context(bin_size=False)
+        if self.env.user.id != SUPERUSER_ID:
+            raise AccessError(
+                _("Encrypted data can only be read as superuser"))
+        if not env:
+            env = self._retrieve_env()
+        encrypted_rec = self.search(
+            [('name', '=', name), ('environment', '=', env)])
+        if not encrypted_rec:
+            return None
+        cipher = self._get_cipher(env)
+        try:
+            return cipher.decrypt(encrypted_rec.encrypted_data).decode()
+        except InvalidToken:
+            raise ValidationError(_(
+                "Password has been encrypted with a different "
+                "key. Unless you can recover the previous key, "
+                "this password is unreadable."))
+
+    @api.model
+    @ormcache('self._uid', 'name', 'env')
+    def _get_json(self, name, env=None):
+        data = self._get(name, env=env)
+        if not data:
+            return {}
+        try:
+            return json.loads(data)
+        except (ValueError, TypeError):
+            raise ValidationError(
+                _("The data you are trying to read are not in a json format"))
+
+    @staticmethod
+    def _retrieve_env():
+        """Return the current environment
+           Raise if none is found
+        """
+        current = config.get('running_env', False)
+        if not current:
+            raise ValidationError(
+                _('No environment found, please check your running_env '
+                  'entry in your config file.'))
+        return current
+
+    @classmethod
+    def _get_cipher(cls, env):
+        """Return a cipher using the key of environment.
+        force_env = name of the env key.
+        Useful for encoding against one precise env
+        """
+        key_name = 'encryption_key_%s' % env
+        key_str = config.get(key_name)
+        if not key_str:
+            raise ValidationError(_(
+                "No '%s' entry found in config file. "
+                "Use a key similar to: %s" % (key_name, Fernet.generate_key())
+            ))
+        # key should be in bytes format
+        key = key_str.encode()
+        return Fernet(key)
+
+    @api.model
+    def _store(self, name, data,  env=None):
+        if self.env.user.id != SUPERUSER_ID:
+            raise AccessError(
+                _("You can only encrypt data as superuser"))
+        if not env:
+            env = self._retrieve_env()
+        cipher = self._get_cipher(env)
```

Comment:
I would split encrypt/decrypt actions to specific methods so that you can easily replace the way it's done. Eg: `_encrypt_data`, `_decrypt_data`.
Not mandatory tho.